src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs

```diff
@@ -522,6 +522,60 @@ internal void ExecutePackageDownload(string name, PackageVersion version, string
                     return;
                 }
 
+                var localPkgs = pmExt.PackageLoader.LocalPackages;
+
+                var uninstallsRequiringRestart = new List<Package>();
+                var uninstallRequiringUserModifications = new List<Package>();
+                var immediateUninstalls = new List<Package>();
+                var stdLibPackages = new List<Package>();
+
+                // if a package is already installed we need to uninstall it, allowing
+                // the user to cancel if they do not want to uninstall the package
+                var duplicateLoacalPackages = package.full_dependency_ids.Select(dep => localPkgs.FirstOrDefault(v => v.Name == dep.name));
```

Comment:
this is the change that fixes part of the duplicate packages problem.
It was broken with this change https://github.com/DynamoDS/Dynamo/pull/10544/files#diff-10009027dea2b20f51c89c7b0306d25ac8d92dbfc0fbfb891fdfa171eba8df1fR707

Reply:
so there is this note in packgeManagerClientViewModel that names will sometimes be null:
https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs#L654

will that now be a problem with this change in comparison?

Reply:
@mmisol can you explain how that can happen ?